Runtime/Scripts/LocationService.cs

```diff
@@ -72,9 +80,26 @@ public static IEnumerator InitalizeLocationService(int maxWait = 20, bool contin
       }
     }
 
-    public IEnumerator Start()
+    public static IEnumerator AssureLocation()
+    {
+         if (Input.location.status == LocationServiceStatus.Initializing)
```

Comment:
I suggest checking if "Input.Location.status != LocationServiceStatus.Running" instead and then checking if LocationServiceStatus is Stopped or Failed, so that we can throw specific exceptions

Reply:
We call Location Once and its being saved


 Awake () > InitializeLocation () > Input.location.start() > location is saved in Input.location.lastData (which is getOnly)

next frame after we get the location , the Input.location.status is LocationServiceStatus.Stopped

Therefore a valid check for location is input.lastData value is not Input.location.status